### PR TITLE
chore: remove refs to deprecated io/ioutil

### PR DESCRIPTION
**Pull request by testwill**

### CODE REVIEW

**Review by olegbespalov — CHANGES_REQUESTED** [review]
Hi @calavera 

Thanks for the pull request! :relaxed: 

Before we can proceed with reviewing it, I'd like to ask you to adjust a few things:

* Ensure that the tests and linters are passed locally by running the following Makefile's targets (`make test` and `make ci-like-lint`)
* Provided a short description to the PR
* Revert a change under the `vendor` folder since it should be done only with the proper module update (`go modules`) .

Thanks!

**Comment by codebien** [comment]
We should also add a `depguard` rule to our `golangci` config.

**Comment by olegbespalov** [comment]
Hey @codebien 

> We should also add a depguard rule to our golangci config.

I briefly skimmed the https://github.com/OpenPeeDeeP/depguard, and wondering if we need it. However, let's open a separate issue (could you please do that since you raised it) where we could discuss a use case.

**Review by imiric — DISMISSED** [review]
Thanks for the PR @testwill! :bow:

It LGTM, but as Oleg mentioned, the lint CI job should pass, and the way we agreed to do that internally is to add `nolint` directives to each case. Since most of these are in tests, we're fine ignoring these specific cases, instead of adding a global rule for all tests.

File: cmd/archive_test.go [hunk 1]
```diff
@@ -77,7 +77,7 @@ func TestArchiveThresholds(t *testing.T) {
 		t.Run(testCase.name, func(t *testing.T) {
 			t.Parallel()
 
-			testScript, err := ioutil.ReadFile(testCase.testFilename)
+			testScript, err := os.ReadFile(testCase.testFilename)
```
Comment (mstoykov) [inline cmd/archive_test.go:80]:
```suggestion
			testScript, err := os.ReadFile(testCase.testFilename) //nolint:forbidigo
```

File: cmd/convert_test.go [hunk 2]
```diff
@@ -102,10 +102,10 @@ export default function() {
 
 func TestConvertCmdCorrelate(t *testing.T) {
 	t.Parallel()
-	har, err := ioutil.ReadFile("testdata/example.har")
+	har, err := os.ReadFile("testdata/example.har")
```
Comment (mstoykov) [inline cmd/convert_test.go:105]:
```suggestion
	har, err := os.ReadFile("testdata/example.har") //nolint:forbidigo
```

File: cmd/convert_test.go [hunk 3]
```diff
@@ -102,10 +102,10 @@ export default function() {
...
 	require.NoError(t, err)
 
-	expectedTestPlan, err := ioutil.ReadFile("testdata/example.js")
+	expectedTestPlan, err := os.ReadFile("testdata/example.js")
```
Comment (mstoykov) [inline cmd/convert_test.go:108]:
```suggestion
	expectedTestPlan, err := os.ReadFile("testdata/example.js") //nolint:forbidigo
```

File: cmd/run_test.go [hunk 4]
```diff
@@ -196,7 +196,7 @@ func TestRunScriptErrorsAndAbort(t *testing.T) {
 		t.Run(name, func(t *testing.T) {
 			t.Parallel()
 
-			testScript, err := ioutil.ReadFile(path.Join("testdata", tc.testFilename))
+			testScript, err := os.ReadFile(path.Join("testdata", tc.testFilename))
```
Comment (mstoykov) [inline cmd/run_test.go:199]:
```suggestion
			testScript, err := os.ReadFile(path.Join("testdata", tc.testFilename)) //nolint:forbidigo
```

File: cmd/run_test.go [hunk 5]
```diff
@@ -251,7 +251,7 @@ func TestInvalidOptionsThresholdErrExitCode(t *testing.T) {
 		t.Run(tc.name, func(t *testing.T) {
 			t.Parallel()
 
-			testScript, err := ioutil.ReadFile(path.Join("testdata", tc.testFilename))
+			testScript, err := os.ReadFile(path.Join("testdata", tc.testFilename))
```
Comment (mstoykov) [inline cmd/run_test.go:254]:
```suggestion
			testScript, err := os.ReadFile(path.Join("testdata", tc.testFilename)) //nolint:forbidigo
```

File: cmd/run_test.go [hunk 6]
```diff
@@ -301,7 +301,7 @@ func TestThresholdsRuntimeBehavior(t *testing.T) {
...
```
Comment (mstoykov) [inline cmd/run_test.go:304]:
```suggestion
			testScript, err := os.ReadFile(path.Join("testdata", tc.testFilename)) //nolint:forbidigo
```

File: js/bundle_test.go [hunk 7]
```diff
@@ -659,7 +658,7 @@ func TestOpen(t *testing.T) {
 			return fs, "", func() {}
 		},
 		"OsFS": func() (fsext.Fs, string, func()) {
-			prefix, err := ioutil.TempDir("", "k6_open_test")
+			prefix, err := os.MkdirTemp("", "k6_open_test")
```
Comment (mstoykov) [inline js/bundle_test.go:661]:
```suggestion
			prefix, err := os.MkdirTemp("", "k6_open_test") //nolint:forbidigo
```

File: js/console_test.go [hunk 8]
```diff
@@ -335,7 +334,7 @@ func TestFileConsole(t *testing.T) {
 						msg, deleteFile := msg, deleteFile
 						t.Run(msg, func(t *testing.T) {
 							t.Parallel()
-							f, err := ioutil.TempFile("", "")
+							f, err := os.CreateTemp("", "")
```
Comment (mstoykov) [inline js/console_test.go:337]:
```suggestion
							f, err := os.CreateTemp("", "") //nolint:forbidigo
```

File: js/runner_test.go [hunk 9]
```diff
@@ -2408,7 +2408,7 @@ func TestComplicatedFileImportsForGRPC(t *testing.T) {
 	}
 
 	fs := fsext.NewMemMapFs()
-	protoFile, err := ioutil.ReadFile("../lib/testutils/httpmultibin/grpc_testing/test.proto")
+	protoFile, err := os.ReadFile("../lib/testutils/httpmultibin/grpc_testing/test.proto")
```
Comment (mstoykov) [inline js/runner_test.go:2411]:
```suggestion
	protoFile, err := os.ReadFile("../lib/testutils/httpmultibin/grpc_testing/test.proto") //nolint:forbidigo
```

**Review by mstoykov — CHANGES_REQUESTED** [review]

**Review by mstoykov — APPROVED** [review]

**Review by olegbespalov — APPROVED** [review]
👍

**Pull request merged**